src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorChangeServiceRequest.java

```diff
@@ -0,0 +1,160 @@
+
+    public static final String SOLICITOR_CHANGE_SERVICE_REQUEST = "solicitor-change-service-request";
+
+
+    @Autowired
+    private ApplicationIssuedNotification applicationIssuedNotification;
+
+    @Autowired
+    private CcdUpdateService ccdUpdateService;
+
+    @Autowired
+    private SetNoticeOfProceedingDetailsForRespondent setNoticeOfProceedingDetailsForRespondent;
+
+    @Autowired
+    private GenerateApplicant1NoticeOfProceeding generateApplicant1NoticeOfProceeding;
+
+    @Autowired
+    private GenerateApplicant2NoticeOfProceedings generateApplicant2NoticeOfProceedings;
+
+    @Autowired
+    private GenerateD10Form generateD10Form;
+
+    @Autowired
+    private GenerateD84Form generateD84Form;
+
+    @Autowired
+    private IdamService idamService;
+
+    @Autowired
+    private AuthTokenGenerator authTokenGenerator;
+
+    @Autowired
+    private NotificationDispatcher notificationDispatcher;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(SOLICITOR_CHANGE_SERVICE_REQUEST)
+            .forStates(POST_SUBMISSION_PRE_AWAITING_CO_STATES)
+            .name("Change service request")
+            .description("Change service request")
+            .showSummary()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grant(CREATE_READ_UPDATE,
+                SOLICITOR)
+            .grantHistoryOnly(CASE_WORKER, LEGAL_ADVISOR, JUDGE))
+            .page("changeServiceRequest")
+            .pageLabel("Change service request")
+            .complex(CaseData::getApplication)
+            .mandatory(Application::getServiceMethod)
+            .showCondition("applicationType=\"soleApplication\"");
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(
+        final CaseDetails<CaseData, State> details,
+        final CaseDetails<CaseData, State> beforeDetails
+    ) {
+        log.info("Solicitor change service request about to submit callback invoked with Case Id: {}", details.getId());
+
+        CaseData caseData = details.getData();
+        final Application application = caseData.getApplication();
+        final boolean isIssued = application.getIssueDate() != null;
+
+        if (application.isPersonalServiceMethod()) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .errors(singletonList("You may not select Personal Service. Please select Solicitor or Court Service."))
+                .build();
+        }
+
+        if (application.isSolicitorServiceMethod() && isIssued) {
+            log.info("Regenerate NOP for App and Respondent for case id: {}", details.getId());
+            final CaseDetails<CaseData, State> solicitorServiceDetails = caseTasks(generateApplicant1NoticeOfProceeding,
```

Comment:
Instead of creating a new casedetails, can we edit the one we have already passed in?